src/applications/disability-benefits/all-claims/utils/submit.js

```diff
@@ -0,0 +1,433 @@
+        if (!ratingKeysToRemove.includes(key)) {
+          acc[key] = disability[key];
+        }
+        return acc;
+      }, {}),
+    );
+  }
+  return clonedData;
+};
+
+/**
+ * Returns an array of the maximum set of PTSD incident form data field names
+ */
+export function getFlatIncidentKeys() {
+  const incidentKeys = [];
+
+  for (let i = 0; i < PTSD_INCIDENT_ITERATION; i++) {
+    incidentKeys.push(`incident${i}`);
+  }
+  for (let i = 0; i < PTSD_INCIDENT_ITERATION; i++) {
+    incidentKeys.push(`secondaryIncident${i}`);
+  }
+
+  return incidentKeys;
+}
+
+export function getPtsdChangeText(changeFields = {}) {
+  return Object.keys(changeFields)
+    .filter(
+      key =>
+        key !== 'other' &&
+        key !== 'otherExplanation' &&
+        PTSD_CHANGE_LABELS[key],
+    )
+    .map(key => PTSD_CHANGE_LABELS[key]);
+}
+
+export function filterServicePeriods(formData) {
+  const { serviceInformation } = formData;
+  if (!serviceInformation || hasGuardOrReservePeriod(serviceInformation)) {
+    return formData;
+  }
+  // remove `reservesNationalGuardService` since no associated
+  // Reserve or National guard service periods have been provided
+  // see https://github.com/department-of-veterans-affairs/va.gov-team/issues/6797
+  const clonedData = _.cloneDeep(formData);
+  delete clonedData.serviceInformation.reservesNationalGuardService;
+  return clonedData;
+}
+
+// Transform the related disabilities lists into an array of strings
+export const stringifyRelatedDisabilities = formData => {
+  if (!formData.vaTreatmentFacilities) {
+    return formData;
+  }
+  const clonedData = _.cloneDeep(formData);
+  const newVAFacilities = clonedData.vaTreatmentFacilities.map(facility =>
+    // Transform the related disabilities lists into an array of strings
+    _.set(
+      'treatedDisabilityNames',
+      transformRelatedDisabilities(
+        facility.treatedDisabilityNames,
+        getClaimedConditionNames(formData, false),
+      ),
+      facility,
+    ),
+  );
+  clonedData.vaTreatmentFacilities = newVAFacilities;
+  return clonedData;
+};
+
+export const cleanUpPersonsInvolved = incident => {
+  // We don't want to require any PTSD entries, but EVSS is rejecting any
+  // personsInvolved that don't have a "injuryDeath" type set. So we're
+  // setting blank entries to "other" and adding a generic other description
+  // see https://github.com/department-of-veterans-affairs/va.gov-team/issues/21422
+  const personsInvolved = incident.personsInvolved?.map((person = {}) => {
+    const name = ['first', 'middle', 'last'].reduce(
+      (fullName, part) => ({
+        ...fullName,
+        [part]: person.name?.[part] || '',
+      }),
+      {},
+    );
+    return person.injuryDeath
+      ? { ...person, name }
+      : {
+          ...person,
+          name,
+          injuryDeath: 'other',
+          injuryDeathOther: person.injuryDeathOther || 'Entry left blank',
+        };
+  });
+  return {
```

Comment:
This is the "actual" code change for this PR. Everything else is moving files around to reduce complexity.